Tribler/Test/Core/Modules/RestApi/Channels/test_channels_torrents_endpoint.py

```diff
@@ -0,0 +1,249 @@
+
+    @deferred(timeout=10)
+    def test_add_torrent_to_channel_500(self):
+        """
+        Testing whether the API returns a formatted 500 error if ValueError is raised
+        """
+        self.create_fake_channel("channel", "")
+        torrent_path = TORRENT_FILE
+
+        def fake_error(channel_id, torrent_def, extra_info=None, forward=True):
+            raise ValueError("Test error")
+
+        self.session.add_torrent_def_to_channel = fake_error
+
+        def verify_error_message(body):
+            error_response = json.loads(body)
+            expected_response = {
+                u"error": {
+                    u"handled": True,
+                    u"code": u"ValueError",
+                    u"message": u"Test error"
+                }
+            }
+            self.assertDictContainsSubset(expected_response[u"error"], error_response[u"error"])
+
+        torrent_file = open(torrent_path, mode='rb')
+        torrent_64 = base64.b64encode(torrent_file.read())
+        post_data = {
+            "torrent": torrent_64
+        }
+        self.should_check_equality = False
+        return self.do_request('channels/discovered/%s/torrents' % 'fakedispersyid'.encode('hex'),
+                               expected_code=500, expected_json=None, request_type='PUT',
+                               post_data=post_data).addCallback(verify_error_message)
+
+
+class TestModifyChannelTorrentEndpoint(AbstractTestChannelsEndpoint):
+
+    def setUp(self, autoload_discovery=True):
+        super(TestModifyChannelTorrentEndpoint, self).setUp(autoload_discovery)
+        self.session.lm.ltmgr = MockObject()
+        self.session.lm.ltmgr.shutdown = lambda: True
+
+    def test_add_torrent_from_url_to_channel_with_description(self):
+        """
+        Testing whether a torrent can be added to a channel using the API
+        """
+        my_channel_id = self.create_fake_channel("channel", "")
+
+        # Setup file server to serve torrent file
+        files_path = os.path.join(self.session_base_dir, 'http_torrent_files')
+        os.mkdir(files_path)
+        shutil.copyfile(TORRENT_FILE, os.path.join(files_path, 'ubuntu.torrent'))
+        file_server_port = get_random_port()
+        self.setUpFileServer(file_server_port, files_path)
+
+        def verify_method_invocation(channel_id, torrent_def, extra_info=None, forward=True):
+            self.assertEqual(my_channel_id, channel_id)
+            self.assertEqual(TorrentDef.load(os.path.join(files_path, 'ubuntu.torrent')), torrent_def)
+            self.assertEqual({"description": "test"}, extra_info or {})
+            self.assertEqual(True, forward)
+
+        self.session.add_torrent_def_to_channel = verify_method_invocation
+
+        # Martijn: We are directly invoking the endpoint here. This is because the requests library used by
+        # TorrentDef.load_from_url is not working well when performing a request using do_request and seems to hang.
```

Comment:
It's probably due to the fact that the `TorrentDef.load_from_url` is a blocking call so I would first like to make it non-blocking and focus on that first (after the wx GUI is removed). After that's done, I can try to convert the lines above back to the 'usual' syntax and see whether it works.